src/software/pipeline/main_panoramaWarping.cpp

```diff
@@ -319,14 +319,18 @@ class CoordinatesMap {
         double dyy = _coordinates(i + 1, j).y() - _coordinates(i, j).y();
 
         double det = std::abs(dxx*dyy - dxy*dyx);
+        if (det < 1e-12) continue;
+        
         scales.push_back(det);
       }
     }
 
     if (scales.size() <= 1) return false;
```

Comment:
```suggestion
    if (scales.empty()) return false;
```
Should work fine when size is 1.